gobblin-docs/sinks/Http.md

```diff
@@ -0,0 +1,105 @@
+  - `ApacheHttpRequestBuilder`. It builds a `ApacheHttpRequest`, which is an `AsyncRequest` that wraps the `HttpUriRequest`, from one `HttpOperation`
+  - `ApacheHttpResponseHandler`. It handles a `HttpResponse`
+
+Configurations for the builder are:
+
+| Configuration | Description | Example
+|---|---|---|
+| `gobblin.writer.http.urlTemplate` | Required, the url template(schema and port included), together with `keys` and `queryParams`, to be resolved to request url | ```http://www.test.com/profiles/${memberId}``` |
+| `gobblin.writer.http.verb` | Required, the type of http request | get, update, delete, etc |
+| `gobblin.writer.http.maxAttempts` | Optional, max number of attempts including initial send | Default is 3 |
+| `gobblin.writer.http.contentType` | Optional, content type of the request body | ```"application/json"```, which is the default value |
+
+## `R2RestWriterBuilder`
+A `R2RestWriterBuilder` builds an `AsyncHttpWriter` on top of [restli r2 framework](https://github.com/linkedin/rest.li/wiki/Request---Response-API-(R2)), sending
+rest request. The 3 major components are:
+
+  - `R2Client`. It uses a R2 [`Client`](https://github.com/linkedin/rest.li/blob/master/r2-core/src/main/java/com/linkedin/r2/transport/common/Client.java) to
+  send [`RestRequest`](https://github.com/linkedin/rest.li/blob/master/r2-core/src/main/java/com/linkedin/r2/message/rest/RestRequest.java) and
+  receive [`RestResponse`](https://github.com/linkedin/rest.li/blob/master/r2-core/src/main/java/com/linkedin/r2/message/rest/RestResponse.java)
+  - `R2RestRequestBuilder`. It builds a `R2Request`, which is an `AsyncRequest` that wraps the `RestRequest`, from one `HttpOperation`
+  - `R2RestResponseHandler`. It handles a `RestResponse`
+  
+ `R2RestWriterBuilder` has [d2](https://github.com/linkedin/rest.li/wiki/Dynamic-Discovery) and ssl support. Configurations(`(d2.)` part should be added in d2 mode) for the builder are:
+ 
+ | Configuration | Description | Example
+ |---|---|---|
+ | `gobblin.writer.http.urlTemplate` | Required, the url template(schema and port included), together with `keys` and `queryParams`, to be resolved to request url. If the schema is `d2`, d2 is enabled | ```http://www.test.com/profiles/${memberId}``` |
+ | `gobblin.writer.http.verb` | Required, the type of http request | get, update, put, delete, etc |
+ | `gobblin.writer.http.maxAttempts` | Optional, max number of attempts including initial send | Default is 3 |
+ | `gobblin.writer.http.d2.zkHosts`| Required for d2, the zookeeper address | |
+ | `gobblin.writer.http.(d2.)ssl`| Optional, enable ssl | Default is false |
+ | `gobblin.writer.http.(d2.)keyStoreFilePath`| Required for ssl | /tmp/identity.p12 |
+ | `gobblin.writer.http.(d2.)keyStoreType`| Required for ssl | PKCS12 |
+ | `gobblin.writer.http.(d2.)keyStorePassword`| Required for ssl | |
```

Comment:
curious question: is it possible to use password manager with this? or the passwords goes in plain text?